### PR TITLE
Quickstart tutorial 

### PR DESCRIPTION
**Pull request by Purvanshsingh**
I have tried to make the Quickstart section of the Documentation, Containing the setup of the Hydrus server using docker-compose.

### CODE REVIEW

**Comment by raoashish10** [comment]
Just a suggestion, I think you can change ```http://localhost:8080/api/vocab#Drone``` to ```http://localhost:8080/api/vocab``` because #Drone is supposed to be specific to the Drone class only even though it returns the whole vocab at the moment.

**Comment by Purvanshsingh** [comment]
> Just a suggestion, I think you can change `http://localhost:8080/api/vocab#Drone` to `http://localhost:8080/api/vocab` because #Drone is supposed to be specific to the Drone class only even though it returns the whole vocab at the moment.

yes, that's a typo thanks will fix it.

File: docs/src/content/quickstart.md [hunk 1]
```diff
@@ -4,4 +4,35 @@ name: Quickstart
 
 # Quickstart Guide
 
-This will lead you to different paths.
+This Quickstart tutorial will guide you to setup the Hydrus server, we will be using docker in this tutorial,
+Therefore you can go through the [Docker container engine](https://docker-curriculum.com/) and its best practices deploying virtualized environments.
```
Comment (priyanshunayan) [inline docs/src/content/quickstart.md:8]: Users don't need to know the best practices to follow this document, right?

File: docs/src/content/quickstart.md [hunk 2]
```diff
@@ -4,4 +4,35 @@ name: Quickstart
...
+
+## Running hydrus server
+> Make sure you have docker-compose installed in your system
```
Comment (priyanshunayan) [inline docs/src/content/quickstart.md:11]: please give links to the official installation guide as well.

File: docs/src/content/quickstart.md [hunk 3]
```diff
@@ -4,4 +4,35 @@ name: Quickstart
...
+```bash
+git clone https://github.com/HTTP-APIs/hydrus
+```
+#### Change directory and run docker-compose
+```bash
+cd hydrus
+
+docker-compose up --build
+```
+> If you are using linux you may have to use sudo command with docker
```
Comment (priyanshunayan) [inline docs/src/content/quickstart.md:23]: > If you are using linux you may have to use sudo command with docker

Is this statement verified?

File: docs/src/content/quickstart.md [hunk 4]
```diff
@@ -4,4 +4,35 @@ name: Quickstart
...
+
+#### Hydrus demo server is up and running at
+```bash
+http://localhost:8080/api/vocab
+```
+
+## Adding your own Hydra Documentation
```
Comment (priyanshunayan) [inline docs/src/content/quickstart.md:30]: In a quick start guide, better to use the default API docs.

File: docs/src/content/quickstart.md [hunk 5]
```diff
@@ -4,4 +4,35 @@ name: Quickstart
...
 # Quickstart Guide
 
-This will lead you to different paths.
+This Quickstart tutorial will guide you to setup the Hydrus server, we will be using docker in this tutorial,
+Therefore you can go through the [Docker container engine](https://docker-curriculum.com/) and its best practices deploying virtualized environments.
+
+## Running hydrus server
+> Make sure you have docker-compose installed in your system
+
+#### Getting the [hydrus repository](https://github.com/HTTP-APIs/hydrus)
```
Comment (priyanshunayan) [inline docs/src/content/quickstart.md:13]: Please maintain a consistent heading hierarchy. If a heading is inside h1, it should be h2. if it is inside h2, it should be h3, not h4.  Inconsistent hierarchy might break the inpage navigation.

File: docs/src/content/quickstart.md [hunk 6]
```diff
@@ -4,4 +4,35 @@ name: Quickstart
...
+```bash
+git clone https://github.com/HTTP-APIs/hydrus
+```
+#### Change directory and run docker-compose
```
Comment (priyanshunayan) [inline docs/src/content/quickstart.md:17]: Having the steps as headings give the impression that there might be some sub-steps as well. I think having an ordered list for the steps can be a good thing. Did you have any special reason to list them as headings?

**Review by priyanshunayan — CHANGES_REQUESTED** [review]
One additional point: A quickstart for an agent too, maybe?

File: docs/src/content/quickstart.md (hunk 1 above)
Comment (Purvanshsingh) [reply docs/src/content/quickstart.md:8]: Yes, you are right my intention was to just make people familiar with docker many of them may not aware of docker-compose.

File: docs/src/content/quickstart.md (hunk 3 above)
Comment (Purvanshsingh) [reply docs/src/content/quickstart.md:23]: https://docs.docker.com/engine/install/linux-postinstall/

The Docker daemon always runs as the root user(default)
we can run it rootless with some extra installations.

**Comment by chrizandr** [comment]
@priyanshunayan merge these commits to a separate branch without review, like we have for the other pages, we can directly edit on the branch then before merging to main. It is difficult to make edits here because we don't have access to Purvansh's fork. Any changes from outside can be made as PRs to the branch.

**Comment by Purvanshsingh** [comment]
I already asked for a separate branch, @priyanshunayan told me that we can go with the master branch itself.

**Pull request merged**